data_steward/cdr_cleaner/cleaning_rules/store_new_duplicate_measurement_concept_ids.py

```diff
@@ -0,0 +1,243 @@
+    and_cn_dupes AS(
+    SELECT
+      dci.value_as_concept_id,
+      dci.n_total_records,
+      dci.n_distinct_measurement_concept_ids,
+      dci.concept_id,
+      dci.concept_name AS concept_name_in_table,
+      dci.vocabulary_id AS concept_vocab_in_table,
+      dci.domain_id,
+      c.concept_id AS dupe_concept_id,
+      c.concept_name,
+      c.vocabulary_id,
+      c.domain_id
+    FROM
+      data_concept_info dci
+    JOIN
+      `{{project_id}}.{{dataset_id}}.concept` c
+    ON
+      dci.concept_name = c.concept_name
+      AND dci.concept_id <> c.concept_id
+    ORDER BY
+      dci.value_as_concept_id ),
+    dedup_values AS (
+    SELECT
+      DISTINCT dupes.value_as_concept_id,
+      dupes.concept_name_in_table AS vac_name,
+      dupes.concept_vocab_in_table AS vac_vocab,
+      COALESCE(CASE
+          WHEN dupes.concept_vocab_in_table NOT IN ('LOINC', 'NAACCR') THEN ( SELECT MAX(value_as_concept_id) FROM and_cn_dupes AS t2 WHERE t2.concept_name_in_table = dupes.concept_name_in_table AND t2.concept_vocab_in_table = 'LOINC' )
+        ELSE
+        dupes.value_as_concept_id
+      END
+        , dupes.value_as_concept_id) AS aou_standard_vac,
+      dupes.n_distinct_measurement_concept_ids AS n_measurement_concept_id,
+      dupes.n_total_records AS n_measurement
+    FROM
+      and_cn_dupes dupes
+    WHERE
+      dupes.dupe_concept_id IN (
+      SELECT
+        DISTINCT value_as_concept_id
+      FROM
+        `{{project_id}}.{{dataset_id}}.measurement`)
+    ORDER BY
+      dupes.concept_name_in_table DESC,
+      dupes.n_total_records DESC,
+      dupes.n_distinct_measurement_concept_ids DESC)
+  SELECT
+    value_as_concept_id,
+    vac_name,
+    vac_vocab,
+    aou_standard_vac,
+    n_measurement_concept_id,
+    n_measurement,
+    CURRENT_DATE() AS date_added
+  FROM
+    dedup_values )
+""")
+
+STORE_NEW_MAPPINGS = JINJA_ENV.from_string("""
+INSERT INTO
+  `{{project_id}}.{{pipeline_dataset}}.{{primare_lookup_table}}` 
+  (value_as_concept_id,
+    vac_name,
+    vac_vocab,
+    aou_standard_vac,
+    date_added)
+SELECT
+  value_as_concept_id,
+  vac_name,
+  vac_vocab,
+  aou_standard_vac,
+  date_added
+FROM
+    `{{project_id}}.{{sandbox_dataset_id}}.{{sandbox_table}}`
+WHERE value_as_concept_id not in (
+  SELECT value_as_concept_id
+  FROM `{{project_id}}.{{pipeline_dataset}}.{{primare_lookup_table}}`)
+""")
+
+
+class StoreNewDuplicateMeasurementConceptIds(BaseCleaningRule):
+    """
+    Store only new occurrences of duplicate lab measurement values.
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id, namer=None):
+        desc = (
+            f'All new duplicate measurement concept_ids will be identified via SQL and '
+            f'stored in a sandbox table.  '
+            f'The table will be read to load into the primary pipeline table,'
+            f'pipeline_tables.IDENTICAL_LABS_LOOKUP_TABLE.')
+
+        namer = dataset_id if not namer else namer
```

Comment:
We might not need this line because it will give `_combined` as table_namer if None by default:
https://github.com/all-of-us/curation/blob/develop/data_steward/cdr_cleaner/cleaning_rules/base_cleaning_rule.py#L432